pkg/controller/populators/import-populator.go

```diff
@@ -183,6 +214,68 @@ func (r *ImportPopulatorReconciler) reconcileTargetPVC(pvc, pvcPrime *corev1.Per
 	return reconcile.Result{}, nil
 }
 
+func CopyEvents(srcObj, destObj client.Object, c client.Client, log logr.Logger, recorder record.EventRecorder) {
+	copyingToDv := false
+	if destObj.GetObjectKind().GroupVersionKind().Kind == "DataVolume" {
+		copyingToDv = true
+	}
+	newEvents := &corev1.EventList{}
+	err := c.List(context.TODO(), newEvents,
+		client.InNamespace(srcObj.GetNamespace()),
+		client.MatchingFields{"involvedObject.name": srcObj.GetName(),
+			"involvedObject.uid": string(srcObj.GetUID())},
+	)
+
+	if err != nil {
+		log.Error(err, "Could not retrieve destObj list of Events")
+	}
+
+	currEvents := &corev1.EventList{}
+	err = c.List(context.TODO(), currEvents,
+		client.InNamespace(destObj.GetNamespace()),
+		client.MatchingFields{"involvedObject.name": destObj.GetName(),
+			"involvedObject.uid": string(destObj.GetUID())},
+	)
+
+	if err != nil {
+		log.Error(err, "Could not retrieve srcObj list of Events")
+	}
+
+	var emitEvent bool
+	for idx, newEvent := range newEvents.Items {
+		emitEvent = true
+		currTime := newEvent.FirstTimestamp.Unix()
+		for _, currEvent := range currEvents.Items {
```

Comment:
Maybe instead of looping over the currEvents, can we put them in a map, and key it on the timestamp. Then if the currEvents already exists, we don't have to emit it. Not sure. Doing 2 loops doesn't seem like the optimal way to go about it. Even if the number of events is pretty low.